app/src/main/java/com/github/drawyourpath/bootcamp/userProfileCreation/PersonalInfoFragment.kt

```diff
@@ -0,0 +1,218 @@
+package com.github.drawyourpath.bootcamp.userProfileCreation
+
+import android.app.DatePickerDialog
+import android.app.Dialog
+import android.graphics.Color
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.DatePicker
+import android.widget.EditText
+import android.widget.TextView
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import com.github.drawyourpath.bootcamp.R
+import com.github.drawyourpath.bootcamp.database.Database
+import com.github.drawyourpath.bootcamp.database.FireDatabase
+import com.github.drawyourpath.bootcamp.database.MockDataBase
+import java.time.LocalDate
+import java.util.*
+
+
+private val MIN_AGE: Int = 10
+private val MAX_AGE: Int = 100
+
+class PersonalInfoFragment : Fragment() {
+    private var isTest: Boolean = false
+    private var userName: String = ""
+    private var dateOfBirth: LocalDate = getTodayDate()
+    private var firstname: String = ""
+    private var surname: String = ""
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view: View = inflater.inflate(R.layout.fragment_personal_info, container, false)
+
+        //retrieve the isRunTestValue and userName from the UserNameTestAndSetFragment
+        val argsFromLastFrag: Bundle? = arguments
```

Comment:
This code appears a lot in the fragments. I think it would be better to create a super-class for all your fragments and add a `getDatabase` method that checks for test flags in the bundle.

Reply:
I think I will modify it later because we will need to modify it to link the authentification with the userId to the user account.